### PR TITLE
Discord deb PR: #360

### PR DESCRIPTION
**Pull request by Itai-Nelken**
for #360

### CODE REVIEW

**Review by fabianmendes — APPROVED** [review]

File: apps/Discord/uninstall [hunk 1]
```diff
@@ -6,6 +6,4 @@ function error {
   echo -e "\\e[91m$1\\e[39m"
   exit 1
 }
-
-gio trash ~/discord-electron
-rm -f ~/.local/share/applications/discord.desktop
```
Comment (SpacingBat3) [inline apps/Discord/uninstall:11]: You might want to leave that for compatibility reasons (users with the old Discord installation might need that when they want to remove it).

**Review by SpacingBat3 — CHANGES_REQUESTED** [review]

File: apps/Discord/uninstall (hunk 1 above)
Comment (Itai-Nelken) [reply apps/Discord/uninstall:11]: good idea

Comment (Itai-Nelken) [reply apps/Discord/uninstall:11]: there, did that

**Comment by Botspot** [comment]
```
Installing Discord with install-32 script
gio: file:///home/pi/discord-electron: Error trashing file /home/pi/discord-electron: No such file or directory
downloading discord-electron...
Failed to download latest release!
Reading package lists... Done
E: Unsupported file /home/pi/electron-discord-webapp_*_armhf.deb given on commandline

Failed to install Discord with install-32 script!
```

**Comment by Itai-Nelken** [comment]
wait, I know what happened. the cide that downloads the latest release is made to be in a function and the be used like this: `function username/repo` but when adding it to the scripts here I forgot to remove that `$1` and change it to `SpacingBat3/electron-discord-webapp`.

**Comment by Itai-Nelken** [comment]
should work now.

**Comment by Botspot** [comment]
Exact same issue still.

```
Installing Discord with install-32 script
gio: file:///home/pi/discord-electron: Error trashing file /home/pi/discord-electron: No such file or directory
downloading discord-electron...
Failed to download latest release!
Reading package lists... Done
E: Unsupported file ./electron-discord-webapp_*_armhf.deb given on commandline
```

Has this code been tested?

File: apps/Discord/install-32 [hunk 2]
```diff
@@ -7,35 +7,11 @@ function error {
...
-Comment=Your place to talk!
-GenericName=Network Messenger" > ~/.local/share/applications/discord.desktop
 
+curl -s https://api.github.com/repos/$1/releases/latest \
+echo "downloading discord-electron..."
+| grep "browser_download_url.*armv7l.deb" \
+| cut -d : -f 2,3 \
+| tr -d \" \
+| wget -qi - | error "Failed to download latest release!"
+sudo apt -f -y install ./electron-discord-webapp_*_armhf.deb
```
Comment (SpacingBat3) [inline apps/Discord/install-32:19]: `apt` doesn't support the wildcards, use `apt-get` instead...

**Review by SpacingBat3 — CHANGES_REQUESTED** [review]

File: apps/Discord/install-32 (hunk 2 above)
Comment (Itai-Nelken) [reply apps/Discord/install-32:19]: so this?
```suggestion
sudo apt-get -f -y install ./electron-discord-webapp_*_armhf.deb
```

**Comment by Itai-Nelken** [comment]
@Botspot I fixed it, I tested it and it works ok.

**Comment by Botspot** [comment]
Now it installs successfully.

However, there is no Menu entry for Discord under the Internet category. Where did it go?

**Comment by Botspot** [comment]
Okay I found it. Now it's under Accessories and named "Electron Discord Web App", instead of "Discord".

Also when I launch it, the menu bar on the top is visible.  
2021-02-17-100915_1920x1080_scrot  
@SpacingBat3 did you have a reason for enabling that? Seems like it cuts down on the realism.

**Comment by Itai-Nelken** [comment]
> Now it installs successfully.
> 
> However, there is no Menu entry for Discord under the Internet category. Where did it go?

I also found that bug, I reported it https://github.com/SpacingBat3/electron-discord-webapp/issues/20

**Comment by Itai-Nelken** [comment]
@Botspot does it work now?
@SpacingBat3 just released 1.0.3 that puts the menu entry under internet and fixes the spellcheck bug.

**Comment by Itai-Nelken** [comment]
@Botspot the scripts work fine, they always download the latest release and install it fine.

**Comment by SpacingBat3** [comment]
@Itai-Nelken You should resolve conflicts in your `pi-apps` repo...

Also, a new version is released with some improvements done to the `*.dekstop` file in both DEB and RPM releases.

**Comment by Itai-Nelken** [comment]
> @Itai-Nelken You should resolve conflicts in your `pi-apps` repo...
> 
> Also, a new version is released with some improvements done to the `*.dekstop` file in both DEB and RPM releases.

@SpacingBat3 I'm not sure I understand what a git conflict is, as far as I understand it happens when the main repo has chages the branch doesn't have and the branch modified the same files.
how can I fix it?

**Comment by Itai-Nelken** [comment]
also my scripts always download the latest release, so no need to change anything.

**Comment by SpacingBat3** [comment]
> > @Itai-Nelken You should resolve conflicts in your `pi-apps` repo...
> > Also, a new version is released with some improvements done to the `*.dekstop` file in both DEB and RPM releases.
> 
> @SpacingBat3 I'm not sure I understand what a git conflict is, as far as I understand it happens when the main repo has chages the branch doesn't have and the branch modified the same files.
> how can I fix it?

See my PR, I've resolved it myself for you if you have no idea how to do that.

**Comment by Itai-Nelken** [comment]
> See my PR, I've resolved it for you if you have no idea how to do that.

thanks! so what you did is simply make a pr from the main repo into my fork?

**Comment by Botspot** [comment]
> @SpacingBat3 I'm not sure I understand what a git conflict is, as far as I understand it happens when the main repo has chages the branch doesn't have and the branch modified the same files.
> how can I fix it?

+1 for zips.

**Comment by SpacingBat3** [comment]
> > See my PR, I've resolved it for you if you have no idea how to do that.
> 
> thanks! so what you did is simply make a pr from the main repo into my fork?

Nah, I just cloned your `pi-apps` repository, added `upstream` to remotes (which is BotSpot's `pi-apps`), `git pull`-ed changes from the `upstream` to my local repo, resolved conflicts with my text editor, commited the changes, pushed them to my own fork and did a PR.

File: apps/Discord/install-32 [hunk 3]
```diff
@@ -21,3 +21,4 @@ rm -f ~/.local/share/applications/discord.desktop 2> /dev/null
 echo "downloading discord-electron..."
 download || error "Failed to download latest release!"
 sudo apt -f -y install ./electron-discord-webapp*armhf.deb
```
Comment (SpacingBat3) [inline apps/Discord/install-32]: AFAIK `apt` **has issues with wildcards**.
However, it [supports *patterns*](https://manpages.debian.org/apt/apt-patterns.7).

If you don't want to use regex nor patterns, you could run `wget` with `-O <output-name>.deb` parameters to download and rename a file to `<output-name>`.

Also, you could use `apt-get` instead, as it supports wildcards. It also skips prompting the user to confirm if he really want to install this package (so there's no need to use `-y` flag.

**Review by SpacingBat3 — CHANGES_REQUESTED** [review]

File: apps/Discord/install-32 (hunk 3 above)
Comment (Itai-Nelken) [reply apps/Discord/install-32]: I used dpkg -i to install the deb (it supports wildcards) nad then `sudo apt -f install` to fix any missing dependencies.

**Review by SpacingBat3 — APPROVED** [review]
I don't think there's anything else broken here.

**Comment by Itai-Nelken** [comment]
@Botspot I see you added the changes, can I close this pr?

**Comment by Botspot** [comment]
There, I've uploaded it with commit 4e867849653228ef83962c24f520ed59cd6515f8

**Pull request closed without merging**